src/Control/Monad/Aff.purs

```diff
@@ -73,15 +73,15 @@ cancelWith aff c = runFn3 _cancelWith nonCanceler aff c
 -- | If you do need to handle exceptions, you can use `runAff` instead, or
 -- | you can handle the exception within the Aff computation, using
 -- | `catchError` (or any of the other mechanisms).
-launchAff :: forall e a. Aff e a -> Eff (err :: EXCEPTION | e) Unit
-launchAff = runAff throwException (const (pure unit)) <<< liftEx
-  where
-  liftEx :: Aff e a -> Aff (err :: EXCEPTION | e) a
-  liftEx = _unsafeInterleaveAff
+launchAff :: forall e a. Aff (err :: EXCEPTION | e) a -> Eff (err :: EXCEPTION | e) (Canceler (err :: EXCEPTION | e))
```

Comment:
To be frank I really struggle to see the utility of the effect record mechanism in Purescript generally.

I understand what you saying in regards to `EXCEPTION` not making sense in `Aff`, in fact this is the first instance I think I've encountered where rejecting certain effects makes sense like this. However `runAff` would still allow you to run an `Aff` with an `EXCEPTION` effect for example.

I think what you actually want is probably `runAff` to be able to declare that the effects in the `Eff` monads are a superset of those in the `Aff` monad but I don't believe that's currently possible in the language. I'm not sure what would be correct for the `Canceler`.

I hadn't thought of `liftEff'` as a kind of dual for `launchAff`; I'll have a think as to whether I can make use of it though I'm concerned by the use of `Either` as opposed to calling the error callback at a glance.

I'll have another stab at it tonight.